designs/node-overlays.md

```diff
@@ -0,0 +1,186 @@
+
+```yaml
+kind: NodeOverlay
+metadata:
+  name: default
+spec:
+  pricePercent: 90
+  overhead:
+    memory: 10Mi
+---
+kind: NodeOverlay
+metadata:
+  name: memory
+spec:
+  weight: 1 # required to resolve conflicts with default
+  selector:
+    matchExpressions:
+      key: karpenter.k8s.aws/instance-memory
+      operator: Gt
+      values: [2048]
+  overhead:
+    memory: 50Mi
+---
+kind: NodeOverlay
+metadata:
+  name: extended-resource
+spec:
+  # weight is not required since default does not conflict with extended resources
+  selector:
+    matchLabels:
+      karpenter.sh/capacity-type: on-demand
+    matchExpressions:
+    - key: node.kubernetes.io/instance-type
+      operator: In
+      values: ["m5.large", "m5.2xlarge", "m5.4xlarge", "m5.8xlarge", "m5.12xlarge"]
+  capacity:
+    smarter-devices/fuse: 1
+```
+
+Option 1 enables us to be strictly more flexible than option 3. However, it introduces ambiguity when multiple NodeOverlays have the same weight. Option 3 would only allow one to apply, so any conflict must be a misconfiguration. However, our example demonstrates that Option 1 has valid use cases where multiple NodeOverlays share the same weight, as long as they don't define the same value. For Option 1, NodeOverlays with the same weight that both attempt to configure the same field are considered to be a misconfiguration.
+
+### Misconfiguration
+
+Given our merge semantics, it's possible for customers to define multiple NodeOverlays with undefined behavior. In both cases, we will identify and communicate these misconfigurations, but should we fail open or fail closed? Failing closed means that a misconfigured would halt Karpenter from launching new Nodes until the misconfiguration is resolved. Failing open means that Karpenter might launch capacity that differs from what an operator expects, and result in thrashing scale in/outs.
+
+We choose to fail closed to prevent undefined behavior that could incur costs. However, we have the option to fail closed for the entire provisioning loop, or for a single NodePool. For example, if multiple NodeOverlays match NodePoolA with the same weight, but only a single NodeOverlay matches NodePoolB, Karpenter could exclude NodePoolA as invalid, but proceed with NodePoolB for simulation. These semantics are very similar to how Karpenter deals with `.spec.limits` or `InsufficientCapacityErrors`. See: https://karpenter.sh/docs/concepts/scheduling/#fallback.
+
+### Observability
+
+How can users be aware of how their NodeOverlays have applied to their NodePools? Following from our decision above, we will model the application of NodeOverlays at the NodePool level. This leads us to observe the behavior of NodeOverlays in the status of the NodePool.
+
+For example:
+```
+kind: NodePool
+status:
+  conditions:
+    - type: InstanceTypesReady
+      status: True
+  nodeOverlays:
+    - default
+    - extended-resources
+  instanceTypes:
```

Comment:
I think the instance types is a good add and something that I would be interested in us adding